Frameworks/CoreGraphics/D2DWrapper.mm

```diff
@@ -19,16 +19,18 @@
 
 using namespace Microsoft::WRL;
 
-// Private helper for creating a D2DFactory
-static ComPtr<ID2D1Factory> __createD2DFactory() {
-    ComPtr<ID2D1Factory> d2dFactory;
-    FAIL_FAST_IF_FAILED(D2D1CreateFactory(D2D1_FACTORY_TYPE_SINGLE_THREADED, __uuidof(ID2D1Factory), &d2dFactory));
-    return d2dFactory;
-}
+HRESULT _CGGetD2DFactory(ID2D1Factory** factory) {
+    static ComPtr<ID2D1Factory> sFactory;
+    static HRESULT sHr;
+    static dispatch_once_t dispatchToken;
+
+    dispatch_once(&dispatchToken, ^{
+        sHr = D2D1CreateFactory(D2D1_FACTORY_TYPE_SINGLE_THREADED, __uuidof(ID2D1Factory), &sFactory);
+    });
 
-ComPtr<ID2D1Factory> _GetD2DFactoryInstance() {
-    static ComPtr<ID2D1Factory> factory = __createD2DFactory();
-    return factory;
+    sFactory.Get()->AddRef();
```

Comment:
I spell real goodly.

---
In reply to: [86616809](https://github.com/Microsoft/WinObjC/pull/1307#discussion_r86616809) [](ancestors = 86616809)